main-service/src/main/java/ru/yandex/model/comment/Comment.java

```diff
@@ -0,0 +1,63 @@
+package ru.yandex.model.comment;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.proxy.HibernateProxy;
+import ru.yandex.model.event.Event;
+import ru.yandex.model.user.User;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "comments")
+public class Comment {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    @ManyToOne
+    @JoinColumn(name = "author_id")
+    private User author;
+    @Column(name = "text")
+    private String text;
+    @Column(name = "created_at")
+    private LocalDateTime createdAt;
+    @Column(name = "changed_at")
+    private LocalDateTime changedAt;
+    @ManyToOne
+    @JoinColumn(name = "event_id")
+    private Event event;
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null) return false;
+        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
+        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
+        if (thisEffectiveClass != oEffectiveClass) return false;
+        Comment comment = (Comment) o;
+        return getId() != null && Objects.equals(getId(), comment.getId());
+    }
+
+    @Override
+    public final int hashCode() {
+        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
+    }
```

Comment:
точно надо?
в принципе можно поставить @EqualsAndHashCode над классом

Reply:
Идея или ампликод ругается, советуя переопределить явно, не через аннотацию. Поменял на аннотацию в итоге

Warning:(29, 1) Using @EqualsAndHashCode for JPA entities is not recommended. It can cause severe performance and memory consumption issues.

Reply:
окей